addon/components/horizontal-bar-chart.js

```diff
@@ -401,6 +405,27 @@ const HorizontalBarChartComponent = ChartComponent.extend(FloatingTooltipMixin,
     return this.get('yAxis').attr(this.get('axisAttrs'));
   },
 
+  _computeBarWidth: function(xScale, value) {
+    return Math.abs(xScale(value) - xScale(0));
+  },
+
+  _checkGroupingLabelOverflow: function(groupingLabelWidth, valueLabelWidth, value, xScale) {
+    const chartSectionWidth = this._computeBarWidth(xScale, value) + valueLabelWidth;
+    return groupingLabelWidth > chartSectionWidth;
+  },
+
+  _computeMarginForGroupingLabel: function(containerWidth, maxGroupingLabelWidth, maxValueLabelWidth, barWidthRatio) {
+    const chartPadding = 2 * this.get('labelPadding') + this._getYAxisTitleOffset();
```

Comment:
Curious what others think cc @Addepar/fire 

Breaking things down into functions is helpful, but does not address my underlying concern, which is that this logic is still very unintuitive, probably hard to maintain if it has bugs, etc.